lib/new_relic/agent/messaging.rb

```diff
@@ -327,12 +328,17 @@ def segment_parameters_enabled?
         NewRelic::Agent.config[:'message_tracer.segment_parameters.enabled']
       end
 
-      def transaction_name(library, destination_type, destination_name)
+      def transaction_name(library, destination_type, destination_name, action = nil)
         transaction_name = Transaction::MESSAGE_PREFIX + library
         transaction_name << NewRelic::SLASH
         transaction_name << Transaction::MessageBrokerSegment::TYPES[destination_type]
         transaction_name << NewRelic::SLASH
 
+        if action == :consume
+          transaction_name << 'Consume'
```

Comment:
Yeah, the reason I didn't do that is mostly because this isn't an rdkafka specific concept. It's new now because rdkafka is the first to do this, but the next kafka gem will also use this, and could also be used for other non kafka related things in the future, and I think using a constant defined in the rdkafka instrumentation would be confusing here.

Reply:
Hmm we could, I don't think it would really make things more clear necessarily or really save anything on performance. It would be different from what we usually do for the instrumentation for these messaging libraries too, but i'm down to make that change if others would prefer it.

Reply:
Since it's different from what we usually do, let's just leave things as-is. Thanks for talking it through with me!